### PR TITLE
Added required methods & getters to use it in forge

### PR DESCRIPTION
**Pull request by dipak-pawar**
#### Short description of what this resolves:
Added changed required to use it in forge.

#### Changes proposed in this pull request:

- Added required methods.
- Added few getters.
- Classloader is not using currentThread.

### CODE REVIEW

File: arquillian-chameleon-container-model/src/main/java/org/arquillian/container/chameleon/spi/model/Container.java [hunk 1]
```diff
@@ -56,5 +68,13 @@ public ContainerAdapter matches(Target target) {
         return null;
     }
 
+    public boolean isVersionMatches(Target target) {
+        if (target.getServer().equalsIgnoreCase(name)) {
+            if (target.getVersion().matches(versionExpression)) {
+                return true;
+            }
+        }
+        return false;
```
Comment (bartoszmajsak) [inline arquillian-chameleon-container-model/src/main/java/org/arquillian/container/chameleon/spi/model/Container.java]: Just wondering, can't we inline it to:

```
return (target.getServer().equalsIgnoreCase(name)) && (target.getVersion().matches(versionExpression))
```

File: arquillian-chameleon-container-model/src/main/java/org/arquillian/container/chameleon/FileUtils.java [hunk 2]
```diff
@@ -72,7 +73,7 @@ private static InputStream loadFileResource(String resourceName) {
     }
 
     private static InputStream loadClassPathResource(String resourceName) {
-        ClassLoader classLoader = Thread.currentThread().getContextClassLoader();
+        ClassLoader classLoader = Target.class.getClassLoader();
```
Comment (bartoszmajsak) [inline arquillian-chameleon-container-model/src/main/java/org/arquillian/container/chameleon/FileUtils.java]: If we found a regression here wouldn't make sense to write a test for it?

Comment (dipak-pawar) [inline arquillian-chameleon-container-model/src/main/java/org/arquillian/container/chameleon/FileUtils.java]: :+1:

File: arquillian-chameleon-container-model/src/test/java/org/arquillian/container/chameleon/FileUtilsTest.java [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+
+public class FileUtilsTest {
+
+    @Rule
+    public ExpectedException thrown= ExpectedException.none();
+
+    @Test
+    public void should_not_load_configuration_file() {
+        thrown.expect(IllegalStateException.class);
+        thrown.expectMessage("Could not find built-in configuration as file nor classloader resource: containers.yaml. Something is terrible wrong with the Classpath.");
```
Comment (bartoszmajsak) [inline arquillian-chameleon-container-model/src/test/java/org/arquillian/container/chameleon/FileUtilsTest.java]: If we really have to say

>  Something is terrible wrong with the Classpath.

then it should be "terribly" I believe.

But to be honest this kind of message is only confusing for the user. Can't we provide some hints, e.g. "make sure this file is in the project folder" etc.

Comment (dipak-pawar) [inline arquillian-chameleon-container-model/src/test/java/org/arquillian/container/chameleon/FileUtilsTest.java]: :+1:

**Comment by bartoszmajsak** [comment]
Make sure to run new formatter from IDEA. Click on the root of the project and  then `ctrl+alt+L`

**Comment by dipak-pawar** [comment]
@bartoszmajsak , I think we can merge this & then I will give PR with all changed due to formatting.

**Comment by bartoszmajsak** [comment]
> I think we can merge this & then I will give PR with all changed due to formatting.

What makes it impossible to do it now?

**Comment by dipak-pawar** [comment]
It has lot of changes. To have more clean PR I am thinking for separate PR

**Comment by bartoszmajsak** [comment]
Upstream @a882313c847922cb96aef5916aaaf136e498cb66 

Thanks!

**Pull request closed without merging**